### PR TITLE
tutorials and demos

### PR DESCRIPTION
**Pull request by jodygarnett**
Collect the JAI tutorial and demo code for distribution under the project Apache License v 2.0.

From jai_demos:

* [x] codec
* [x] demo
* [x] medical
* [x] mpv
* [x] network

From jai_tutorial:

* [x] docs/tutorial 

This commit is focused on providing the files (compiling with maven to confirm nothing was missed).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Good news - with the most recent update I am able to run the first demo application:

image

**Comment by jodygarnett** [comment]
Please note this branch requires Java 8 to compile, have not yet addressed the need to compile on Java 11.

File: docs/migration/index.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   Replacing:
+   
+   ```XML
+   <properties>
+      <jai.version>0.4-SNAPSHOT</jai.version>
+   </properties>
+   ...
+   <dependency>
+     <groupId>org.eclipse.imagen</groupId>
+     <artifactId>imagen_core</artifactId>
```
Comment (jnh5y) [inline docs/migration/index.md]: The new package is presently named imagen-core.

File: docs/migration/index.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     <version>${jai.version}</version>
+   </dependency>
+   <dependency>
+     <groupId>org.eclipse.imagen</groupId>
+     <artifactId>jai_codec</artifactId>
```
Comment (jnh5y) [inline docs/migration/index.md]: Does this exist?

**Comment by jodygarnett** [comment]
I have received check-in permission via CQ19766 and this pull request is now merged.

**Pull request merged**